src/Zope2/utilities/tests/test_zconsole.py

```diff
@@ -0,0 +1,79 @@
+import os
+import sys
+import tempfile
+import unittest
+
+if sys.version_info.major == 2:
+    from StringIO import StringIO
+else:
+    from io import StringIO
+
+zope_conf_template = """
+%define INSTANCE {}
+
+instancehome $INSTANCE
+
+<zodb_db main>
+   <filestorage>
+     path $INSTANCE/Data.fs
+   </filestorage>
+   mount-point /
+</zodb_db>
+"""
+
+test_script = """
+import sys
+
+
+if __name__ == '__main__':
+    app.foo = '42'
+    print(app.foo)
+    print(sys.argv[1:])
+"""
+
+
+class ZConsoleTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.instancedir = tempfile.mkdtemp(prefix='instance')
```

Comment:
I'd like if if the temp dir gets deleted in `tearDown`.

Reply:
+1 and I actually tried doing this, however then I get:

```
Error in test test_database_events (Zope2.App.tests.test_startup.StartupTests)
FileNotFoundError: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/tmp/instancep8089in_/Data.fs.lock'
```

which to me looks like an unrelated test isolation problem.